test/object-store/main.cpp

```diff
@@ -57,7 +59,128 @@ int main(int argc, char** argv)
+        Base::testCaseStarting(testCaseInfo);
+    }
+    void testCaseEnded(TestCaseStats const& testCaseStats) override
+    {
+        auto it = m_results.find(testCaseStats.testInfo.name);
+        if (it == m_results.end()) {
+            throw std::runtime_error("logic error in Evergreen section reporter");
+        }
+        if (testCaseStats.totals.assertions.allPassed()) {
+            it->second.status = "pass";
+        }
+        else {
+            it->second.status = "fail";
+        }
+        it->second.end_time = std::chrono::system_clock::now();
+        Base::testCaseEnded(testCaseStats);
+    }
+    void sectionStarting(SectionInfo const& sectionInfo) override
+    {
+        if (m_pending_name.empty()) {
+            m_pending_name = sectionInfo.name;
+        }
+        else {
+            m_pending_name += "::" + sectionInfo.name;
+        }
+        m_pending_test = {};
+        Base::sectionStarting(sectionInfo);
+    }
+    void sectionEnded(SectionStats const& sectionStats) override
+    {
+        if (!m_pending_name.empty()) {
+            if (sectionStats.assertions.allPassed()) {
+                m_pending_test.status = "pass";
+            }
+            else {
+                m_pending_test.status = "fail";
+            }
+            m_pending_test.end_time = std::chrono::system_clock::now();
+            m_results.emplace(std::make_pair(m_pending_name, m_pending_test));
+            m_pending_name = "";
+        }
+        Base::sectionEnded(sectionStats);
+    }
+    void testRunEndedCumulative() override
+    {
+        auto results_arr = nlohmann::json::array();
+        for (const auto& [test_name, cur_result] : m_results) {
+            auto to_millis = [](const auto& tp) -> double {
+                return static_cast<double>(
+                    std::chrono::duration_cast<std::chrono::milliseconds>(tp.time_since_epoch()).count());
+            };
+            double start_secs = to_millis(cur_result.start_time) / 1000;
+            double end_secs = to_millis(cur_result.end_time) / 1000;
+            int exit_code = 0;
+            if (cur_result.status != "pass") {
+                exit_code = 1;
+            }
+
+            nlohmann::json cur_result_obj = {{"test_file", test_name}, {"status", cur_result.status},
+                                             {"exit_code", exit_code}, {"start", start_secs},
+                                             {"end", end_secs},        {"elapsed", end_secs - start_secs}};
+            results_arr.push_back(std::move(cur_result_obj));
+        }
+        auto result_file_obj = nlohmann::json{{"results", std::move(results_arr)}};
+        stream << result_file_obj << std::endl;
+    }
+
+    struct TestResult {
+        TestResult()
+            : start_time{std::chrono::system_clock::now()}
+            , end_time{}
+            , status{"unknown"}
+        {
+        }
+        std::chrono::time_point<std::chrono::system_clock> start_time;
```

Comment:
I think you can do std::chrono::system_clock::time_point here to save some typing.